tests/reader-config.rs

```diff
@@ -42,3 +42,134 @@ mod expand_empty_elements {
+
+    #[test]
+    fn false_() {
+        let mut reader = Reader::from_str(XML);
+        reader.trim_text(false);
+
+        assert_eq!(
+            reader.read_event().unwrap(),
+            Event::Text(BytesText::new(" \t\r\n"))
+        );
+
+        assert_eq!(
+            reader.read_event().unwrap(),
+            Event::DocType(BytesText::new("root \t\r\n"))
+        );
+        assert_eq!(
+            reader.read_event().unwrap(),
+            Event::Text(BytesText::new(" \t\r\n"))
+        );
+
+        assert_eq!(
+            reader.read_event().unwrap(),
+            Event::Start(BytesStart::from_content("root \t\r\n", 4))
+        );
+        assert_eq!(
+            reader.read_event().unwrap(),
+            Event::Text(BytesText::new(" \t\r\n"))
+        );
+
+        assert_eq!(
+            reader.read_event().unwrap(),
+            Event::Empty(BytesStart::from_content("empty \t\r\n", 5))
+        );
+        assert_eq!(
+            reader.read_event().unwrap(),
+            Event::Text(BytesText::new(" \t\r\ntext \t\r\n"))
+        );
+
+        assert_eq!(
+            reader.read_event().unwrap(),
+            Event::Comment(BytesText::new(" comment \t\r\n"))
+        );
+        assert_eq!(
+            reader.read_event().unwrap(),
+            Event::Text(BytesText::new(" \t\r\n"))
+        );
+
+        assert_eq!(
+            reader.read_event().unwrap(),
+            Event::CData(BytesCData::new(" \t\r\ncdata \t\r\n"))
+        );
+        assert_eq!(
+            reader.read_event().unwrap(),
+            Event::Text(BytesText::new(" \t\r\n"))
+        );
+
+        assert_eq!(
+            reader.read_event().unwrap(),
+            Event::PI(BytesText::new("pi \t\r\n"))
+        );
+        assert_eq!(
+            reader.read_event().unwrap(),
+            Event::Text(BytesText::new(" \t\r\n"))
+        );
+
+        assert_eq!(
+            reader.read_event().unwrap(),
+            Event::End(BytesEnd::new("root"))
+        );
+        assert_eq!(
+            reader.read_event().unwrap(),
+            Event::Text(BytesText::new(" \t\r\n"))
+        );
+
+        assert_eq!(reader.read_event().unwrap(), Event::Eof);
+    }
+
+    #[test]
+    fn true_() {
+        let mut reader = Reader::from_str(XML);
+        reader.trim_text(true);
+
+        assert_eq!(
+            reader.read_event().unwrap(),
+            Event::DocType(BytesText::new("root \t\r\n"))
+        );
+        assert_eq!(
+            reader.read_event().unwrap(),
+            Event::Start(BytesStart::from_content("root \t\r\n", 4))
+        );
+        assert_eq!(
+            reader.read_event().unwrap(),
+            Event::Empty(BytesStart::from_content("empty \t\r\n", 5))
+        );
+        assert_eq!(
+            reader.read_event().unwrap(),
+            Event::Text(BytesText::new("text"))
+        );
+        assert_eq!(
+            reader.read_event().unwrap(),
+            Event::Comment(BytesText::new(" comment \t\r\n"))
```

Comment:
Do we have any reason to support trimming the text values of "comments"?  I cannot immediately think of a reason to do that, but perhaps one exists.

Reply:
I think not, at least nobody request such feature. If such a request appears, we can add a separate option.

Generally speaking, I would delete the current trim options as they simply do not work correctly for text alternating with CDATA / comments / processing instructions, but I suppose that would break many users. I was thinking about renaming current `Event` into `RawEvent` and `DeEvent` to `Event` and give users stream of `Event`s. The `RawEvent` then would be a low-level event which usually not needed by most users. That is very raw thoughts currently, so I decided to not do revolutional changes for now.